### PR TITLE
Application CRUD commands

### LINKED ISSUE #45: Command to create, list, show, update, delete, suspend, and resume applications
Lower level command for application management

### PR DESCRIPTION
**Pull request by eguzki**
New commands implemented:
- [x] application list
- [x] application create
- [x] application show
- [x] appliation apply
- [x] application delete
- [x] application suspend
- [x] application resume

Fixes #45 

Missing integration tests

### CODE REVIEW

**Comment by eguzki** [comment]
ready for review @miguelsorianod

File: docs/applications.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+* `application` positional argument is application unique identifier. Allowed id's are:
+  * Application internal id
+  * User_key (API key)
+  * App_id (from app_id/app_key pair)
+  * Client ID (for OAuth and OpenID Connect authentication modes)
+
+```shell
+NAME
+    delete - delete application
```
Comment (miguelsorianod) [inline docs/applications.md]: Missing options subsection?

File: lib/3scale_toolbox/commands/application_command/apply_command.rb [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+              description <<-HEREDOC
+              Update (create if it does not exist) application'
+              \n Application param allows:
+              \n * Application internal id
+              \n * User_key (API key)
+              \n * App_id (from app_id/app_key pair)
+              \n * Client ID (for OAuth and OpenID Connect authentication modes)
+              HEREDOC
+
+              option      nil, 'user-key', 'User Key (API Key) of the application to be created.', argument: :required
```
Comment (miguelsorianod) [inline lib/3scale_toolbox/commands/application_command/apply_command.rb]: shouldn't this be a symbol as with the other ones?

File: lib/3scale_toolbox/commands/application_command/apply_command.rb [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+              option      nil, 'application-key', 'App Key(s) or Client Secret (for OAuth and OpenID Connect authentication modes) of the application to be created. Only used when application does not exist.' , argument: :required
```
Comment (miguelsorianod) [inline lib/3scale_toolbox/commands/application_command/apply_command.rb]: shouldn't this be a symbol as with the other ones?

File: lib/3scale_toolbox/commands/application_command/create_command.rb [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+          def self.command
+            Cri::Command.define do
+              name        'create'
+              usage       'create [opts] <remote> <account> <service> <application-plan> <name>'
+              summary     'create one application'
+              description 'create one application linked to given account and application plan'
+
+              option      nil, 'user-key', 'User Key (API Key) of the application to be created.', argument: :required
+              option      nil, 'application-id', 'App ID or Client ID (for OAuth and OpenID Connect authentication modes) of the application to be created. ', argument: :required
```
Comment (miguelsorianod) [inline lib/3scale_toolbox/commands/application_command/create_command.rb]: same thing about the symbol as in other commands

File: docs/applications.md (hunk 1 above)
Comment (eguzki) [reply docs/applications.md]: left out deliberately.

Does not add any value IMO

File: lib/3scale_toolbox/commands/application_command/apply_command.rb (hunk 2 above)
Comment (eguzki) [reply lib/3scale_toolbox/commands/application_command/apply_command.rb]: not necessarily

File: lib/3scale_toolbox/commands/application_command/apply_command.rb (hunk 3 above)
Comment (eguzki) [reply lib/3scale_toolbox/commands/application_command/apply_command.rb]: not necessarily

File: lib/3scale_toolbox/commands/application_command/create_command.rb (hunk 4 above)
Comment (eguzki) [reply lib/3scale_toolbox/commands/application_command/create_command.rb]: not necessarily

File: lib/3scale_toolbox/commands/application_command/apply_command.rb (hunk 2 above)
Comment (eguzki) [reply lib/3scale_toolbox/commands/application_command/apply_command.rb]: https://github.com/ddfreyne/cri/blob/master/samples/sample_simple.rb#L24

**Comment by eguzki** [comment]
changes applied

**Pull request merged**